### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by slembcke**
Fix Mac build issues. Fix flipped Mac scrolling issue. Fix issue with tableview visible rows.

### CODE REVIEW

File: cocos2d-ui/CCScrollView.m [hunk 1]
```diff
@@ -1171,9 +1169,12 @@ - (void)scrollWheel:(NSEvent *)theEvent
     {
         // Update scroll position
         CGPoint scrollPos = self.scrollPosition;
-        scrollPos = ccpAdd(delta, scrollPos);
+        scrollPos.x += delta.x;
+        scrollPos.y = (_flipYCoordinates ? -scrollPos.y : scrollPos.y) + delta.y;
```
Comment (slembcke) [inline cocos2d-ui/CCScrollView.m:1173]: I think this is probably a workaround for a different bug. Scroll positions go from [0, -height] for flipped scroll views. I think this is changed from the 3.x behavior and probably a bug.

**Comment by slembcke** [comment]
Tested and works with SpriteBuilder.

File: cocos2d-ui/CCScrollView.m (hunk 1 above)
Comment (andykorth) [inline cocos2d-ui/CCScrollView.m:1173]: This is consistent with other changes I made regarding _flipYCoordinates.

Comment (slembcke) [inline cocos2d-ui/CCScrollView.m:1173]: The scrollPosition property is public though. It seems weird to set a negative scroll position.

Comment (andykorth) [inline cocos2d-ui/CCScrollView.m:1173]: Oh yeah, that property used to flip the position.

**Pull request merged**